js/src/api/rpc/parity/parity.js

```diff
@@ -95,6 +95,11 @@ export default class Parity {
       .execute('parity_dappsList');
   }
 
+  dappsRefresh () {
```

Comment:
Yeah, I think refreshing the page is actually ok when you add a new dapp. As you said it will happen once a month at most.
IMHO not worth supporting additional RPC method and this additional JS code.

But I don't have very strong opinion on that. Since the code is already there we can probably merge it.